qp_shogun/sortmerna/sortmerna.py

```diff
@@ -0,0 +1,180 @@
+                        '--aligned %s --other %s '
+                        '--fastx --blast 1 --num_alignments 1 -a %s' % (
+                            RNA_REF_DB, f_fp,
+                            join(out_dir, '%s.ribosomal.R1' % run_prefix),
+                            join(out_dir, '%s.nonribosomal.R1' % run_prefix),
+                            threads))
+
+            cmds.append('sortmerna --ref %s --reads %s '
+                        '--aligned %s --other %s '
+                        '--fastx --blast 1 --num_alignments 1 -a %s' % (
+                            RNA_REF_DB, r_fp,
+                            join(out_dir, '%s.ribosomal.R2' % run_prefix),
+                            join(out_dir, '%s.nonribosomal.R2' % run_prefix),
+                            threads))
+    return cmds, samples
+    # In this version I have not added a summary file or sam file
+
+
+def sortmerna(qclient, job_id, parameters, out_dir):
+    """Run Sortmerna with the given parameters
+    Parameters
+    ----------
+    qclient : tgp.qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values
+    out_dir : str
+        The path to the job's output directory
+    Returns
+    -------
+    bool, list, str
+        The results of the job
+    """
+    # Step 1 get the rest of the information need to run Sortmerna
+    qclient.update_job_step(job_id, "Step 1 of 4: Collecting information")
+    artifact_id = parameters['input']
+    del parameters['input']
+
+    # Get the artifact filepath information
+    artifact_info = qclient.get("/qiita_db/artifacts/%s/" % artifact_id)
+    fps = artifact_info['files']
+
+    # Get the artifact metadata
+
+    prep_info = qclient.get('/qiita_db/prep_template/%s/'
+                            % artifact_info['prep_information'][0])
+    qiime_map = prep_info['qiime-map']
+
+    # Step 2 generating command for Sortmerna
+    qclient.update_job_step(job_id, "Step 2 of 4: Generating"
+                                    " SortMeRNA commands")
+    rs = fps['raw_reverse_seqs'] if 'raw_reverse_seqs' in fps else []
+    commands, samples = generate_sortmerna_commands(fps['raw_forward_seqs'],
+                                                    rs, qiime_map, out_dir,
+                                                    parameters)
+
+    # Step 3 executing Sortmerna
+    len_cmd = len(commands)
+    msg = "Step 3 of 4: Executing ribosomal filtering (%d/{0})".format(len_cmd)
+    success, msg = _run_commands(qclient, job_id,
+                                 commands, msg, 'QC_Sortmerna')
+    if not success:
+        return False, None, msg
+
+    ainfo = []
+
+    # Generates 2 artifacts: one for the ribosomal
+    # reads and other for the non-ribosomal reads
+    # Step 4 generating artifacts for Nonribosomal reads
+
+    msg = "Step 4 of 5: Generating artifacts "
+    "for Nonribosomal reads (%d/{0})".format(len_cmd)
```

Comment:
I'm surprised this was not flagged by flake8.

```suggestion
    msg = ("Step 4 of 5: Generating artifacts "
           "for Nonribosomal reads (%d/{0})").format(len_cmd)
```